metric/recorder.go

```diff
@@ -0,0 +1,11 @@
+package metric
+
+// Recorder initialize all metrics
+func Recorder() {
+	registerMetrics()
```

Comment:
why do we need a function that only call registerMetrics and not calling it directly ?  It would be nice instead this Recorder become an interface so we can easily mock it for tests

Reply:
Well, I am not sure if we are not going to need more operations regarding configuring all the metrics system. For me now `metrics.go` is where the real metrics stuff is done and this `Recorder` is responsible for interacting with the subsystem. As you mention here, I can make it an interface so we abstract for testing purposes all metrics system, so although now I agree that it is so simple class, it helps for testing purposes and in the future might (or not) help on register phase.

Reply:
yeah i believe this would be needed to be an interface if we want to mock this out in the tests as this will be more and more integrated into the code,

Reply:
Doing it now.